src/main/java/de/tum/in/www1/artemis/repository/DynamicSpecificationRepository.java

```diff
@@ -0,0 +1,64 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import jakarta.persistence.criteria.JoinType;
+import jakarta.validation.constraints.NotNull;
+
+import org.springframework.data.jpa.domain.Specification;
+
+import de.tum.in.www1.artemis.domain.DomainObject_;
+import de.tum.in.www1.artemis.repository.fetchOptions.FetchOptions;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * A repository that provides methods for fetching entities dynamically using {@link FetchOptions}.
+ *
+ * @param <T> the type of the entity
+ */
+public interface DynamicSpecificationRepository<T> {
```

Comment:
```suggestion
public interface DynamicSpecificationRepository<T, Key> extends JpaRepository<T, Key> {
```
I think every subclass of this interface will also implement the JpaRepository, so we could simplify the inheritance of the other Repositories